### PR TITLE
Added async storage and return bike page

### PR DESCRIPTION
**Pull request by lyanita**
Pull in Kristin's PR to reuse the bike item component.

I added async storage to allow our app to store the uid (and other key-value pairs if we like), so we can reference which user's data we need to grab.

To test it out on the return bike page, you can change out the rider value in the testRides.tsx file with your uid (it will be logged to your console when you log in, or you take a look at what's already stored in Firebase Authentication/Firestore).

The return bike page currently uses mock data, then filters to the current ride and therefore bike.

### CODE REVIEW

File: client/pages/Login.tsx [hunk 1]
```diff
@@ -2,61 +2,14 @@ import React, {useState, useEffect} from 'react';
 import { KeyboardAvoidingView, StyleSheet, Text, TextInput, TouchableOpacity, View } from 'react-native';
 import { useTheme } from '@react-navigation/native';
 import { useNavigation } from '@react-navigation/core';
+import AsyncStorage from '@react-native-async-storage/async-storage';
```
Comment (jonramm) [inline client/pages/Login.tsx:5]: Very cool library! I've been grabbing _uid_  from the _auth_ object but this is maybe an easier way to have it accessible everywhere.

File: client/pages/ReturnBike.tsx [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+    const getBike = async () => {
+        var result = bikes.filter(function(val) {
+            return (val["bike_id"] == ride[0].bike);
+        });
+        console.log(result);
+        setBike(result);
+    }
+
+    useEffect(() => {
+        getUid().then(uid => {
```
Comment (jonramm) [inline client/pages/ReturnBike.tsx:46]: I've been getting 'Possible Unhandled Promise Rejection' errors in my console and undefined variables due to this chain of calls. Also, in line 40 for example, shouldn't the _uid_ after the _getUid().then_ be the return value from _getUid()_? Maybe returning each successive data item down the line makes it eaier to manage?

File: client/pages/ReturnBike.tsx [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+                getBike();
+            });
+        });
+    }, []);
+
+    // if a ride is checked out by the user
+    if (ride.length > 0) {
+        return (
+            <View style={styles.bikesContainer}>
+                <Text>Trip in Progress</Text>
```
Comment (jonramm) [inline client/pages/ReturnBike.tsx:57]: I'm seeing the 'Trip in Progress' text but not the BikeItem component when I run it

**Review by jonramm — CHANGES_REQUESTED** [review]
As noted below, I'm getting 'Possible Unhandled Promise Rejection' errors when I run the app that are blocking the display of the BikeItem component. Awesome job with the AsyncStorage library though, that'll be a great way to store items that we need quick access to everywhere in the app.

File: client/pages/ReturnBike.tsx (hunk 2 above)
Comment (lyanita) [reply client/pages/ReturnBike.tsx:46]: My intent was that the uid state variable would update first, then the ride state variable then the bike state variable. I've changed it so it returns values instead and passes them onto the next function as a param.

Comment (jonramm) [reply client/pages/ReturnBike.tsx:46]: Cool! I'm always getting stuck between state variables, return values, and so-called 'callback hell' so I'm not really sure which method is better.

**Review by jonramm — DISMISSED** [review]
Cool, that totally works for me now!

File: client/data/testRides.tsx [hunk 4]
```diff
@@ -0,0 +1,20 @@
+const rides = [
+    {
+        'ride_id': '1234',
+        'start_time': '2023-01-01 00:00:00', // Placeholder: Pretty sure firestore doesn't store timestamps this way
```
Comment (schaefkr) [inline client/data/testRides.tsx:4]: You might be able to use the firestore Timestamp class: firebase.firestore.Timestamp.now()

File: client/pages/Login.tsx (hunk 1 above)
Comment (schaefkr) [reply client/pages/Login.tsx:5]: Sounds neat. I will need to spend some time learning about how to use this.

**Review by schaefkr — APPROVED** [review]

**Pull request merged**